### DISCUSSION TITLE
Wishlist items - UI configuration or clean up

### DISCUSSION THREAD

**Discussion by Pletoo** [sheltermanager/asm3#1515] [Ideas]
Rather than making a separate post for each one, here are some wishlist items that seem like they would be fairly simple to add, but probably aren't.

### Messages:

- Restrict the ability to send messages to "everyone" based on user roles. Further fine tuning who a user could message would be even better.
- Give an option to set whether real names (preferably first name, last initial) usernames, or user roles (preferably specific roles rather than all or none) are shown in the message "FOR" dropdown list.
### Reports:

- Report Query builder - clean up the "Fields", "Criteria" and "Sort" dropdown menus. For example, there are 18 fields that relate to ActiveMovement, Could they be nested or collapsed to make finding the right field categories easier?
- Currently if a role is not set, everyone can see that report. An option to change the default behavior to only show reports specified for their user role would be helpful. It would be faster than assigning roles to every single report to restrict who sees it. 
- Ability to see if a report was installed or created (and by whom) when looking at the full list on the edit reports screen. _(A report created by a user should always be visible to that user, overriding the user role setting.)_
### Shelter view:

- A way to configure the sort list. For example: we don't have adoption coordinators and we only have cats. I'd like to clear out some of those excess sort options that don't apply to us.

### Emblems:

- Customize even the default emblems. For example: the fostered and adoptable emblems look very similar, especially when using the desktop interface on a phone with a small screen, so I'd like to change one of them.

### Logs:

- Ability to set default log type through user roles or user settings rather than only a system default.
- Ability to lock a log type to a user role.
### Homepage:

- Add information about recently uploaded animal media (for example, documents scanned in via the mobile UI) and logs (with the ability to set which type logs appeared here) similar to the diary and/or message boards. Since adding these to an animal record does not cause them to receive the recently changed status, this would make it easy to see when these have been added and provide a link to them.

**Comment by bobintetley** [comment]
Thanks, all good feedback. There are reports to view recently added media in the repository - eg: View forms/documents received between two dates. There's a variant that runs for "today" that can be emailed as well as variants for photos. They have links to the related record so you can click straight into it.

You can install reports from the repository via Settings->Reports->Browse sheltermanager.com

**Reply by Pletoo** [reply]
Thank you. The document repository and the report builder are both excellent features that I use heavily, and I have spent quite a bit of time in the report library so I appreciate those as well. (I've actually explored so much that I have forgotten which reports I have installed and which I have created and now I need to clean a bunch out, hence the request to see who installed or created a report while on the edit reports screen.)

I had tried some of the reports using a search for media or documents. The "View all Media Links" and the "View forms/documents attached between two dates" reports both returned with no data to show. I just discovered that I should have searched for PDF and photo because I found what I was looking for.

**Comment by Pletoo** [comment]
On the littermates screen, location for a deceased cat is shown using the death category. 
![image](https://github.com/user-attachments/assets/8548b0c5-e3c0-4808-862d-7a22f73f938c)

That is helpful when looking for trends, but for clarity, could the word "Deceased" be added like it is in the search results?
![image](https://github.com/user-attachments/assets/429e175d-24c9-43ac-bb69-0e60db46e49d)

**Comment by Pletoo** [comment]
### Bulk Change Animals

- If an item has been removed via options (such as the "good with" fields) give an option to also have it removed from the bulk change screen.
- Allow our custom "additional fields" to be changed in bulk through this screen as well.
### User Settings

- Add Quicklinks for Document Repository and Bulk Change Animals.

### Alerts

- When an animal changes age groups. (Our adoption donations are based on age group and length of time on shelter.)

**Comment by bobintetley** [comment]
Thanks, good suggestions, I will add tickets for these.

**Comment by Pletoo** [comment]
An option to choose which sections from the "Notes" tab is displayed up top. It gets really messy when there are multiple notes.
![notes](https://github.com/user-attachments/assets/5cab97a4-d7cb-44b4-87b0-77540fdf1365)

To keep them grouped together on Shelter View, we give kittens a name like TN/CC/01 which is foster initials/mother initials/# until they are old enough to go to the vet, then we name them. I added an additional field called AKA to keep track of the coded name for records but putting it in the Hidden Comments section is really nice when the name is first changed so that it is more visible, but having the other info there too kind of defeats the point of putting it there. It would be really nice to be able to prioritize which one of those fields is displayed (for example, display Hidden Comments if it has a value otherwise display Markings.) 
![image](https://github.com/user-attachments/assets/02411a1e-dea3-414a-8b2c-d0e1619833c4)

**Comment by Pletoo** [comment]
### Quicklinks
Similar to the recently deceased (a good alternative to having them on the homepage)
- Recently Adopted (maybe include transferred too)
- Recently Added (maybe include those returned from adoption? doesn't necessarily need ones coming from foster.)

### Shelter view sort

- Name no foster
- Litter and location (show litters first, then the others by location)

### User Settings

- An option to choose to remember last shelter view sort until log out. 

-

**Comment by bobintetley** [comment]
The "Move->Return an animal from adoption" screen shows animals who were recently adopted (last 30 days). A recently added search keyword / shelterview mode is a great idea.

**Reply by Pletoo** [reply]
On my tablet, the Quicklink to Recently Deceased produces a nicely formatted screen (I assume because it is a search result?) 
![image](https://github.com/user-attachments/assets/2451225a-d6ed-43cf-9cf0-c6031a146266)

The Return from Adoption produces a mess, requiring a scroll.
![image](https://github.com/user-attachments/assets/96e2f4d2-86d2-481d-953e-17954764e7a6) !

As I set up user accounts and roles, I also have been signing on to that account and adding quicklinks relevant to that user (rather than using the system default.)  The quicklinks are visible, easy access and they don't have to remember which menu they found something under, but I also want to know what they see when they use the link and if there is anything they should not see so I also test things. 

Everyone wants to see the recently adopted - it is why we do what we do. I would like to give them a location filter of Adopted along with the view animals permission without also having to give them the view movements permission. Then give them a quicklink to click.

**Comment by bobintetley** [comment]
Part of that is because table views display poorly on mobile. We're about to release a big fix for that this week to make the desktop UI much more accessible on mobile. I will add a similar keyword for recently adopted to recently deceased to the ticket. The timeline, which I think you removed, is a good source of this info in both interfaces?

**Reply by Pletoo** [reply]
I'm glad to hear of the fix coming out, and thank you for the additional quicklinks. I appreciate being able to make things as easy as possible for my co-workers and Fosterers. I know it is a extra work for you for something that already exists in another form.

The timeline was a little overwhelming on the mobile, I have it as a quicklink on the account but I don't think my co-workers use it and they haven't asked about it missing from the homepage.

I have been using computers pretty much all my life and I am pretty comfortable learning new software. When I first started in the office and got access to SM, I tried the mobile UI, and went immediately back to the desktop UI mainly because of the timeline. The first additional user (young guy) I added did the same thing. It just isn't what we are looking for when we first log on. Months later, as I explored this very configurable software, I took another look at the UI wondering if it might work for our fosters. I haven't tried the limited foster user accounts with the timeline, so I will take a look, but unless it is now in the burger menu as well, I probably won't add it back in.

I have been working with one of our fosterers (who also happens to be a board member, accountant, and owner of the master user account who has used the desktop interface for 5+ years), when she logs in on the mobile UI - her instinct is not to click on the burger menu but to tap on the 'everyone' that is immediately visible with the message function. I hear her mumble that she doesn't want to see everyone - just her cats. then I direct her to the burger menu. She will get there eventually, but finding the cat info and photo uploader under animals within the burger menu really is not coming naturally to her.

I told her to keep the master user account for when she is using the desktop and then gave her a 2nd account bookmarked on her phone, so I could tie in her fosters, and we will work on limiting that second account so that she sees only the info she wants to on the mobile, chances are pretty good it won't be the timeline.

It seems to work best for new users when menu items are visible but the info is hidden away until needed. It is why, as I previously mentioned, I do like the older interface better simply because the view animals and photo uploader were pretty intuitive being on the main screen, However, I get the toolkit no longer being supported and having to move on. It will just take a little longer for me to train and for them to get comfortable, but we will get there.

**Comment by bobintetley** [comment]
Sure, I can appreciate that for the first time a user encounters something new, but users should be able to learn what something means or where it lives if it's a piece of software they're using all the time!

**Reply by Pletoo** [reply]
They will.
I plan to introduce a little at a time.
